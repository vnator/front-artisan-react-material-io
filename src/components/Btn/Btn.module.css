.Btn {
  margin: var(--space-large) 0;
  font-size: var(--size-base);
  border: 1px solid var(--white-smoke);
  height: 36px;
  display: flex;
  align-items: center;
  justify-content: center;
  box-sizing: border-box;
  background: var(--white-glass);
  color: var(--black-smoke);
  transition: var(--transition-base);
  cursor: pointer;
  width: 100%;
}

.Btn:hover {
  background: var(--white-full);
  color: var(--black-full);
}
