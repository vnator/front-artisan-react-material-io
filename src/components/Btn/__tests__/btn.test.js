import React from 'react';
import { Btn } from '../Btn';
import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { spy } from 'sinon';

Enzyme.configure({ adapter: new Adapter() });

describe('Btn', () => {
  it('Snapshot', () => {
    const btn = mount(<Btn onClick={() => console.log('test')}>test</Btn>);

    expect(btn).toMatchSnapshot();
  });

  it('onClick', () => {
    const btnSpy = spy();
    const btn = shallow(<Btn onClick={btnSpy}>test</Btn>);

    btn.simulate('click');

    expect(btnSpy.calledOnce).toBeTruthy();
  });
});
