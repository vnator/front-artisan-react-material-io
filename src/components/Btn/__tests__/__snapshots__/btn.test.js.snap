// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Btn Snapshot 1`] = `ReactWrapper {}`;
